### PR TITLE
Validate user-methods parameters

### LINKED ISSUE #890: Add argument checks for methods in user-methods.js
As is nothing happens which is confusing. We should probably throw an error for an invalid number of arguments. I saw this behavior in node.

### PR DESCRIPTION
**Pull request by kristiandupont**
Resolves #890

### CODE REVIEW

File: lib/user-methods.js [hunk 1]
```diff
@@ -24,6 +24,19 @@ function node_require(module) {
     return require(module);
 }
 
+function typeOf(obj) {
+    return ({}).toString.call(obj).match(/\s(\w+)/)[1].toLowerCase();
```
Comment (fealebenpae) [inline lib/user-methods.js]: How about `Object.prototype.toString`? Or using the `typeof` operator?

Comment (kristiandupont) [inline lib/user-methods.js]: I don't know what I was thinking here -- something about types not being available in the arguments array, which doesn't make any sense..

File: lib/user-methods.js [hunk 2]
```diff
@@ -24,6 +24,15 @@ function node_require(module) {
     return require(module);
 }
 
+function checkTypes(args, types) {
+    args = [].slice.call(args);
```
Comment (fealebenpae) [inline lib/user-methods.js]: This should be `Array.prototype.slice`.

Comment (kristiandupont) [inline lib/user-methods.js]: I guess you are right :)

**Pull request merged**